### PR TITLE
Add docs on CLI and memos

### PR DESCRIPTION
**Pull request by dmetzgar**
Added memos to the key features page. Added CLI doc pulled from the readme in the server repo. Modified CLI doc to include info on starting workflows with memos as well as some other small edits.

### CODE REVIEW

File: docs/08_cli.md [hunk 1]
```diff
@@ -0,0 +1,202 @@
+# Command Line Interface
+
+The Cadence CLI is a command-line tool to perform various tasks on a Cadence server. It can perform 
+domain operations such as register, update, and describe as well as workflow operations like start 
+workflow, show workflow history, and signal workflow.
+
+## Using the CLI
+
+The Cadence CLI can be used directly from the Docker Hub image *ubercadence/cli* or by building the CLI tool
```
Comment (vancexu) [inline docs/08_cli.md]: Do you think it worth to make CLI available in open homebrew?

File: docs/08_cli.md [hunk 2]
```diff
@@ -0,0 +1,202 @@
...
+./cadence workflow start --tl helloWorldGroup --wt main.Workflow --et 60 -i '"cadence"' --wid "<duplicated workflow id>" --wrp 0
+
+# use AllowDuplicate option to run a workflow
+./cadence workflow run --tl helloWorldGroup --wt main.Workflow --et 60 -i '"cadence"' --wid "<duplicated workflow id>" --wrp 1
+```
+
+##### Start workflow with a memo
+Memos are immutable key/value pairs that can be attached to a workflow run when starting the workflow. These are 
+visible when listing workflows. More information on memos can be found 
+[here]({{ '/docs/02_key_features#durable-timers' | relative_url }}).
```
Comment (vancexu) [inline docs/08_cli.md]: The link looks not correct

File: docs/08_cli.md (hunk 1 above)
Comment (dmetzgar) [reply docs/08_cli.md]: Ya, that would be nice. Docker method would still be needed on Windows but at least Mac users would have a better experience.

File: docs/08_cli.md (hunk 2 above)
Comment (dmetzgar) [reply docs/08_cli.md]: Good catch

File: docs/08_cli.md [hunk 3]
```diff
@@ -0,0 +1,202 @@
...
+```
+
+#### Start workflow
+```
+./cadence workflow start --tl helloWorldGroup --wt main.Workflow --et 60 -i '"cadence"'
+
+# view help messages for workflow start
+./cadence workflow start -h
+
+# for workflow with multiple input, separate each json with space/newline like
```
Comment (purplesusan) [inline docs/08_cli.md]: for workflow with multiple input -> for a workflow with multiple inputs

File: docs/08_cli.md [hunk 4]
```diff
@@ -0,0 +1,202 @@
...
+./cadence workflow start --tl helloWorldGroup --wt main.WorkflowWith3Args --et 60 -i '"your_input_string" 123 {"Name":"my-string", "Age":12345}'
+```
+The workflow `start` command is similar to the `run` command, but immediately returns the workflow_id and 
+run_id after starting the workflow. Use the `show` command to view the workflow's history/progress.  
+
+##### Re-use the same workflow id when starting/running workflow
```
Comment (purplesusan) [inline docs/08_cli.md]: Re-use -> Reuse
starting/running workflow -> starting/running a workflow

File: docs/08_cli.md [hunk 5]
```diff
@@ -0,0 +1,202 @@
...
+
+Use option `--workflowidreusepolicy` or `--wrp` to configure the workflow id re-use policy.
```
Comment (purplesusan) [inline docs/08_cli.md]: re-use -> reuse

File: docs/08_cli.md [hunk 6]
```diff
@@ -0,0 +1,202 @@
...
+**Option 2 RejectDuplicate:** Do not allow starting a workflow execution using the same workflow ID as a previous workflow.  
+```
+# use AllowDuplicateFailedOnly option to start a workflow
+./cadence workflow start --tl helloWorldGroup --wt main.Workflow --et 60 -i '"cadence"' --wid "<duplicated workflow id>" --wrp 0
+
+# use AllowDuplicate option to run a workflow
+./cadence workflow run --tl helloWorldGroup --wt main.Workflow --et 60 -i '"cadence"' --wid "<duplicated workflow id>" --wrp 1
+```
+
+##### Start workflow with a memo
```
Comment (purplesusan) [inline docs/08_cli.md]: Start workflow -> Start a workflow

File: docs/08_cli.md [hunk 7]
```diff
@@ -0,0 +1,202 @@
...
+# a shortcut of this is (without -w -r flag)
+./cadence workflow showid 3ea6b242-b23c-4279-bb13-f215661b4717 866ae14c-88cf-4f1e-980f-571e031d71b0
+
+# if run_id is not provided, it will show the latest run history of that workflow_id
+./cadence workflow show -w 3ea6b242-b23c-4279-bb13-f215661b4717
+# a shortcut of this is
+./cadence workflow showid 3ea6b242-b23c-4279-bb13-f215661b4717
+```
+
+#### Show workflow execution info
```
Comment (purplesusan) [inline docs/08_cli.md]: info -> information

File: docs/08_cli.md [hunk 8]
```diff
@@ -0,0 +1,202 @@
...
+# default will only show one page, to view more items, use --more flag
+./cadence workflow list -m
+```
+
+#### Query workflow execution
+```
+# use custom query type
+./cadence workflow query -w <wid> -r <rid> --qt <query-type>
+
+# use build-in query type "__stack_trace" which is supported by cadence client library
```
Comment (purplesusan) [inline docs/08_cli.md]: cadence -> Cadence

**Pull request merged**